cmd/gardener-extension-shoot-flux/app/options.go

```diff
@@ -0,0 +1,154 @@
+	RESTConfig     *rest.Config
+	ManagerOptions manager.Options
+}
+
+// newOptions creates a new options instance.
+func newOptions() *options {
+	opts := &options{
+		generalOptions: &extensionscmdcontroller.GeneralOptions{},
+		restOptions:    &extensionscmdcontroller.RESTOptions{},
+		managerOptions: &extensionscmdcontroller.ManagerOptions{
+			// These are default values.
+			LeaderElection:          true,
+			LeaderElectionID:        extensionscmdcontroller.LeaderElectionNameID(fluxv1alpha1.ExtensionType),
+			LeaderElectionNamespace: os.Getenv("LEADER_ELECTION_NAMESPACE"),
+			MetricsBindAddress:      ":8080",
+			HealthBindAddress:       ":8081",
+		},
+		extensionOptions: &extensionscmdcontroller.ControllerOptions{
+			// This is a default value.
+			MaxConcurrentReconciles: 5,
+		},
+		healthOptions: &extensionscmdcontroller.ControllerOptions{
+			// This is a default value.
+			MaxConcurrentReconciles: 5,
+		},
+		heartbeatOptions: &extensionsheartbeatcmd.Options{
+			ExtensionName: fluxv1alpha1.ExtensionType,
+			Namespace:     os.Getenv("LEADER_ELECTION_NAMESPACE"),
+		},
+		controllerSwitches: extensionscmdcontroller.NewSwitchOptions(
+			extensionscmdcontroller.Switch(extension.ControllerName, extension.AddToManager),
+			extensionscmdcontroller.Switch(extensionshealthcheckcontroller.ControllerName, healthcheck.AddToManager),
+			extensionscmdcontroller.Switch(extensionsheartbeatcontroller.ControllerName, extensionsheartbeatcontroller.AddToManager),
+		),
+		reconcileOptions: &extensionscmdcontroller.ReconcilerOptions{},
+	}
+
+	opts.optionAggregator = extensionscmdcontroller.NewOptionAggregator(
+		opts.generalOptions,
+		opts.restOptions,
+		opts.managerOptions,
+		extensionscmdcontroller.PrefixOption(extension.ControllerName+"-", opts.extensionOptions),
+		extensionscmdcontroller.PrefixOption(extensionshealthcheckcontroller.ControllerName+"-", opts.healthOptions),
+		extensionscmdcontroller.PrefixOption(extensionsheartbeatcontroller.ControllerName+"-", opts.heartbeatOptions),
+		opts.controllerSwitches,
+		opts.reconcileOptions,
+	)
+
+	return opts
+}
+
+func (o *options) addFlags(fs *pflag.FlagSet) {
+	o.optionAggregator.AddFlags(fs)
+}
+
+func (o *options) Complete() error {
+	if err := o.optionAggregator.Complete(); err != nil {
+		return err
+	}
+
+	// customize rest config
+	o.RESTConfig = o.restOptions.Completed().Config
+	// TODO: consider dropping this or make these settings configurable
+	util.ApplyClientConnectionConfigurationToRESTConfig(&componentbaseconfig.ClientConnectionConfiguration{
+		QPS:   100.0,
+		Burst: 130,
+	}, o.RESTConfig)
+
+	// customize manager options
+	o.ManagerOptions = o.managerOptions.Completed().Options()
+	o.ManagerOptions.WebhookServer = nil // this extension doesn't run webhooks
+	o.ManagerOptions.GracefulShutdownTimeout = ptr.To(5 * time.Second)
+
+	o.ManagerOptions.Client.Cache = &client.CacheOptions{
+		DisableFor: []client.Object{
+			&corev1.Secret{}, // applied for ManagedResources
```

Comment:
> // applied for ManagedResources

what does that mean?

Reply:
This is a leftover, good catch

Reply:
We do not apply secrets for `ManagedResources`, but we read the referenced secret.
We probably don't want to watch and cache all secrets on the seed, though.

Reply:
makes sense - `Get()` starts the informer cache implicitly...